### PR TITLE
Correspondences

### PR DESCRIPTION
**Pull request by yosefm**
Take the correspondences code from openptv-python. Break it down. Clean it up. Test it. Fix it. 

Fixes to notice: 
(a) old code would permanently mark a target used even when the candidate clique failed on a target from another camera.
(b) reinstated consistent pairs.
(c) Properly stop generating candidates if scratch buffer is full. This used to segfault.

Bindings (to be done properly) will take some more time, as I need to think about the best way to do it Pythonically. I already used it in my own UI and it looks beautiful, but my bindings are not ready for prime-time. So let's focus on the C code for now.

### CODE REVIEW

**Comment by alexlib** [comment]
15: Running suite(s): Testing correspondences
15: Overflow in correspondences.
15: 88%: Checks: 9, Failures: 0, Errors: 1
15: /Users/alex/Documents/OpenPTV/alexlib/liboptv/tests/check_correspondences.c:445:E:Calibration target correspondences:test_correspondences:0: (after this point) Received signal 10 (Bus error: 10)
15/15 Test #15: check_correspondences ............***Failed    0.05 sec

**Comment by yosefm** [comment]
I didn't change anything there. It worked as usual for me. Can you reproduce it? Say, install a previous version and then try make verify again?

**Comment by alexlib** [comment]
in the previous version I could run `make verify` . It is possible that if there is already `liboptv` installed on the machine, it's somehow used during `make verify`. If this is a case we'd need to inform the user to remove the installed version and run `make verify` again?

File: liboptv/src/correspondences.c [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+        parameters.
+*/
+void match_pairs(correspond *list[4][4], coord_2d **corrected, 
+    frame *frm, volume_par *vpar, control_par *cpar, Calibration **calib) 
+{
+    int i1, i2, i, j, pt1, count;
+    double xa12, ya12, xb12, yb12; /* Epipolar line edges */
+    candidate cand[MAXCAND];
+    
+    for (i1 = 0; i1 < cpar->num_cams - 1; i1++) {
```
Comment (alexlib) [inline liboptv/src/correspondences.c]: is it possible to do this bi-directional? wouldn't it improve substantially the quality of the correspondences?

**Comment by alexlib** [comment]
I reviewed the code. It's well done. I recommend to merge

**Comment by alexlib** [comment]
I can reproduce the failure of the last test if only `make` and `make verify`. what I do is install the version from `openptv/master` (including `sudo make install`) and then checkout the new branch, run again: 
```
 cmake ../
make 
make verify
```

File: liboptv/src/correspondences.c (hunk 1 above)
Comment (yosefm) [inline liboptv/src/correspondences.c]: Yes, but it would require significant attention and testing. Better leave it for later.

**Comment by yosefm** [comment]
In this case, I think 'make clean' will solve your problem. 

I see that 'make' did not build any file in liboptv which is suspicious. Switching branches from master to this one should have triggered a whole bunch of building.

**Pull request merged**